lib/tictactoe.rb

```diff
@@ -1,42 +1,78 @@
+require 'ai_basic'
+require 'ai_minimax'
 require 'game_board'
 require 'game_rules'
+require 'game_state'
 require 'player_manager'
-require 'game_loop'
-require 'game_screen'
-class TicTacToe
 
-  attr_accessor :board, :game_loop
+class TicTacToe
 
-  def initialize (game_io, player1, player2)
+  def initialize
```

Comment:
Darek, 
I could have started by making a web app or terminal app that just used a mock of the TicTacToe gem. Then after completing that I could have then wrote the gem file. 

I decided to go with the bottom up approach because I already had the tictactoe code written. Also, I had never made a gem before so I kind of wanted to attempt doing that. I did notice that, I was spending time trying to figure out what methods I would need to provide for an application that didn't exist yet. This also made me worry that I might be forgetting something, or that I was writing methods that would never actually be needed. 

I think what I would like to do moving forward, is to forget about this branch, try a top down approach and work on the web version and terminal version of the game, and after completing those create a separate branch off of the master branch and create the gem again with the knowledge of knowing what methods a program using it would expect to have available.